ompi/group/group.c

```diff
@@ -495,7 +495,7 @@ int ompi_group_compare(ompi_group_t *group1,
     bool similar, identical;
     ompi_group_t *group1_pointer, *group2_pointer;
     ompi_proc_t *proc1_pointer, *proc2_pointer;
-
+    opal_mutex_lock (&ompi_proc_lock); 
```

Comment:
Any reason why the `OPAL_THREAD_LOCK()/UNLOCK()` macro wasn't used?

Reply:
Thanks for pointing this out Austen, fixed it.